packages/core/src/utils/Error.ts

```diff
@@ -608,6 +612,246 @@ canvas {
   }
 };
 
+export const errLocs = (
```

Comment:
Another way would be to add this to an umbrella type so every error is required to identify the offending source location. This is also fine, but the downside is yet another place to update when a new error is added.